test/qa/lib/fileshare.py

```diff
@@ -34,24 +37,28 @@ def create_directory(file_count: int, name_suffix: str = "", parent_dir: str | N
         name_suffix (str, optional): A suffix to append to the filenames. Defaults to an empty string.
         parent_dir (str | None, optional): The parent directory where the temporary directory will be created.
                                            If None, the system default temporary directory is used. Defaults to None.
-        file_size (str, optional): The size of each file to be created, specified using typical file size notation
-                                   (e.g., "1K", "128M"). Defaults to "1K".
+        file_size (int, optional): The size of each file to be created, in megabytes. Default: `1` MB; Maximum: `1024` MB.
     Returns:
         Directory: A Directory object containing:
             - dir_path: Path to the created directory.
             - paths: Full paths to the created files.
             - transfer_paths: File paths with leading directories removed.
             - filenames: Names of the created files.
     """
+
+    if file_size > MAX_FILE_SIZE:
+        ex = f"Maximum allowed file size is {MAX_FILE_SIZE} MB"
+        raise ValueError(ex)
+
+    exec_command = CommandExecutor(ssh_client)
```

Comment:
Could we name this something like `ssh_command` or `ssh_exec` or something similar? Now if you want to figure out if it's executed locally or via ssh, you need to find the declaration and check if `ssh_client` was passed in or not